force-app/main/default/lwc/ldsDeleteRecord/__tests__/ldsDeleteRecord.test.js

```diff
@@ -0,0 +1,110 @@
+import { createElement } from 'lwc';
+import LdsDeleteRecord from 'c/ldsDeleteRecord';
+import { deleteRecord } from 'lightning/uiRecordApi';
+import { registerApexTestWireAdapter } from '@salesforce/lwc-jest';
+import getAccountList from '@salesforce/apex/AccountController.getAccountList';
+
+// Realistic data with a list of contacts
+const mockGetAccountList = require('./data/getAccountList.json');
+// An empty list of records to verify the component does something reasonable
+// when there is no data to display
+const mockGetAccountListNoRecords = require('./data/getAccountListNoRecords.json');
+
+// Register as an Apex wire adapter. Some tests verify that provisioned values trigger desired behavior.
+const getAccountListAdapter = registerApexTestWireAdapter(getAccountList);
+
+describe('c-lds-delete-record', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    describe('getAccountList @wire data', () => {
+        it('renders two records with name and lightning-button-icon', () => {
+            const ACCOUNT_NAME_RESULT = 'GenePoint';
+            const ACCOUNT_ID_RESULT = '0011700000r5uciAAA';
+            const ACCOUNT_COUNT = 7;
```

Comment:
Let's extract these 3 values from the imported data instead of hardcoding them. That will make the test resilient to data content updates as long as the shape remains. I'd also update the names to make it clear it's data for the first account.

```
const FIRST_ACCOUNT_NAME = mockGetAccountList[0].Name;
const FIRST_ACCOUNT_ID = mockGetAccountList[0].Id;
const ACCOUNT_COUNT = mockGetAccountList.length;
```

Reply:
Will add later. There are many PRs open right now... Doing this, plus adding more comments, will be a general cleanup task.